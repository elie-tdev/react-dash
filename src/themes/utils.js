export const opacity = (hex, opacity) => {
  var result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
  if (!result) {
    throw new Error('Invalid hex', hex);
  }
  return `rgba(${parseInt(result[1], 16)}, ${parseInt(result[2], 16)}, ${parseInt(
    result[3],
    16,
  )}, ${opacity})`;
};

export const gradient = ({ type = 'linear', colors, background }) =>
  `${type}-gradient(${colors.join(', ')}), ${background};`;
