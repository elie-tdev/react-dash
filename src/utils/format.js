const formatter = new Intl.NumberFormat('en-US', {
  style: 'currency',
  currency: 'USD',
  minimumFractionDigits: 2,
});

const parseFloat = (str, val = 2) => {
  str = str && str.toString();
  str = str && str.indexOf('.') > -1 ? str.slice(0, str.indexOf('.') + val + 1) : str;
  return Number(str);
};

export const formatDollar = number => formatter.format(parseFloat(number, 2));
