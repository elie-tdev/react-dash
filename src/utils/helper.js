const maskAccountNumber = accNumber => {
  return accNumber && <>&bull;&bull;&bull;&bull; {accNumber.slice(accNumber.length - 4)}</>;
};

export default maskAccountNumber;
