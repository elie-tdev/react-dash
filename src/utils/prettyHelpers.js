export const prettyCurrency = (amount, currency = 'USD') => {
  if (amount == null) {
    return false;
  }
  if (amount < 0) {
    amount = 0;
  }
  return amount.toLocaleString('en', { style: 'currency', currency });
};

export const prettyDateTime = time => {
  if (time == null) {
    return false;
  }
  let d = new Date(time);
  return (
    d.toLocaleDateString('en', {
      month: '2-digit',
      day: '2-digit',
      year: '2-digit',
    }) +
    ' at ' +
    d.toLocaleTimeString('en', { hour: '2-digit', minute: '2-digit' }).replace(/^0(?:0:0?)?/, '')
  );
};

export const prettyDate = time => new Date(time).toLocaleDateString('en', {});

export const prettyTime = time =>
  new Date(time).toLocaleTimeString([], { hour: '2-digit', minute: '2-digit' });

export const prettyHour = time =>
  new Date(time)
    .toLocaleTimeString('en', { hour: '2-digit', minute: '2-digit' })
    .replace(/^0(?:0:0?)?/, '');

export const prettyPhone = number =>
  number.substr(2).replace(/(\d{1,2})(\d{1})?(\d{1,3})?(\d{1,4})?/, function (_, p1, p2, p3, p4) {
    let output = '';
    if (p1) output = `${p1}`;
    if (p2) output += `${p2}-`;
    if (p3) output += `${p3}-`;
    if (p4) output += `${p4}`;
    return output;
  });

export const minimumDays = (date, numDays) => {
  const d = new Date(date);
  const totalNumDays = daysIntoYear(d) + numDays;
  return Boolean(daysIntoYear(d) >= totalNumDays || totalNumDays <= 365);
};

export const daysIntoYear = date => {
  return (
    (Date.UTC(date.getFullYear(), date.getMonth(), date.getDate()) -
      Date.UTC(date.getFullYear(), 0, 0)) /
    24 /
    60 /
    60 /
    1000
  );
};

export const cleanAmountRegex = value => {
  return value.replace(/[^0-9]/g, '');
};

export const cleanAmount = value => {
  const pos = value.indexOf('.');
  let result;
  if (pos !== -1) {
    const part1 = value.substr(0, pos);
    const part2 = value.substr(pos + 1);
    result = cleanAmountRegex(part1) + '.' + cleanAmountRegex(part2);
  } else {
    result = cleanAmountRegex(value);
  }
  return result;
};
