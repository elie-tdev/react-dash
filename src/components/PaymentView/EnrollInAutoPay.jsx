import cn from 'classnames';
import React, { useState } from 'react';
import { Card, makeStyles, Typography, Box } from '@material-ui/core';
import { useTranslation } from 'react-i18next';

import autopayCardBackground from '@/static/images/autopayCardBackground.svg';
import { gradient, useGlobalStyles } from '@/themes';
import { PrimaryButton, SubmitPaymentModal } from '@/components';

const useStyles = makeStyles(theme => ({
  card: {
    minHeight: '240px',
    background: gradient({
      type: 'radial',
      colors: theme.palette.common.gradient.primary.dark,
      background: theme.palette.primary.main,
    }),
  },
  box: {
    display: 'flex',
    justifyContent: 'space-between',
    flexDirection: 'column',
    alignItems: 'flex-start',
    padding: '24px',
    height: '100%',
    minHeight: 'inherit',
    backgroundImage: `url(${autopayCardBackground})`,
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'center center',
    backgroundSize: '100% 100%',
  },
  title: {
    width: '75%',
    fontWeight: 600,
    fontSize: '22px',
    lineHeight: '32px',
    color: theme.palette.primary.contrastText,
    [theme.breakpoints.down('md')]: {
      width: '100%',
    },
  },
  button: {
    height: 'auto',
  },
}));

const EnrollInAutoPay = ({ onEnrollInAutoPay }) => {
  const { t } = useTranslation();
  const styles = useStyles();
  const globalStyles = useGlobalStyles();
  const [isAutoPaySetup, setIsAutoPaySetup] = useState(false);

  return (
    <Card className={cn(globalStyles.card, styles.card)}>
      <Box className={cn(styles.box)}>
        <Typography variant="h1" className={styles.title}>
          {t('content.Enroll in AutoPay and never miss a payment again')}
        </Typography>
        <PrimaryButton
          variant="text"
          fullWidth={false}
          onClick={() => setIsAutoPaySetup(true)}
          className={styles.button}
        >
          {t('content.Enroll in AutoPay')}
        </PrimaryButton>
      </Box>
      {isAutoPaySetup && (
        <SubmitPaymentModal
          open={isAutoPaySetup}
          isAutoPaySetup={isAutoPaySetup}
          onClose={() => {
            setIsAutoPaySetup(false);
          }}
          onSuccess={onEnrollInAutoPay}
        />
      )}
    </Card>
  );
};

export default EnrollInAutoPay;
