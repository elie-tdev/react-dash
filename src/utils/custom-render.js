import React from 'react';
import { I18nextProvider } from 'react-i18next';
import { render } from '@testing-library/react';
import { CssBaseline } from '@material-ui/core';
import { BrowserRouter as Router } from 'react-router-dom';
import { ThemeProvider } from '@material-ui/core/styles';
import { MockedProvider } from '@apollo/client/testing';

import { MenuContextProvider } from '@/contexts';
import i18n from '@/translations';
import { LightTheme } from '@/themes';

const wrapper = ({ mocks = [], menuContextProviderProps = {} }) => {
  const Wrapper = ({ children }) => {
    return (
      <Router>
        <MockedProvider mocks={mocks} addTypename={false}>
          <ThemeProvider theme={LightTheme}>
            <MenuContextProvider {...menuContextProviderProps}>
              <CssBaseline />
              <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
            </MenuContextProvider>
          </ThemeProvider>
        </MockedProvider>
      </Router>
    );
  };

  return Wrapper;
};

const customRender = (ui, { mocks, menuContextProviderProps, ...restOptions } = {}) =>
  render(ui, { wrapper: wrapper({ mocks, menuContextProviderProps }), ...restOptions });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
