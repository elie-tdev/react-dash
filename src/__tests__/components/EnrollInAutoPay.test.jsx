import React from 'react';
import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { render } from '@/utils/custom-render';
import EnrollInAutoPay from '@/components/PaymentView/EnrollInAutoPay';

describe('EnrollInAutoPay Component', () => {
  test('renders EnrollInAutoPay', () => {
    const mock = jest.fn();
    const { container } = render(<EnrollInAutoPay onEnrollInAutoPay={mock} />);
    userEvent.click(screen.getByText('Enroll in AutoPay'));
    expect(mock.mock.calls).toHaveLength(0);
    expect(container).toMatchSnapshot();
  });
});
