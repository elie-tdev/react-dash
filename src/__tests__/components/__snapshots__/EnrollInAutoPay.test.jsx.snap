// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnrollInAutoPay Component renders EnrollInAutoPay 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-card-8 makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiBox-root MuiBox-root-34 makeStyles-box-2"
    >
      <h1
        class="MuiTypography-root makeStyles-title-3 MuiTypography-h1"
      >
        Enroll in AutoPay and never miss a payment again.
      </h1>
      <button
        class="MuiButtonBase-root MuiButton-root makeStyles-root-35 makeStyles-root-36 MuiButton-text makeStyles-button-4 MuiButton-textSizeLarge MuiButton-sizeLarge MuiButton-disableElevation"
        data-testid="PrimaryButton"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Enroll in AutoPay
        </span>
        <span
          class="MuiTouchRipple-root"
        >
          <span
            class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
            style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
          >
            <span
              class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
            />
          </span>
        </span>
      </button>
    </div>
    <div
      aria-hidden="true"
      class="MuiBackdrop-root makeStyles-backdrop-32"
      style="opacity: 1; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; webkit-transform: none; transform: none;"
    >
      <div
        class="MuiDrawer-root makeStyles-root-50 MuiDrawer-modal"
        id="submitPaymenModalt"
        role="presentation"
        style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
      >
        <div
          aria-hidden="true"
          class="MuiBackdrop-root"
          style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        />
        <div
          data-test="sentinelStart"
          tabindex="0"
        />
        <div
          class="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorBottom makeStyles-modalPaper-33 MuiPaper-elevation16"
          data-testid="SubmitPayment"
          tabindex="-1"
        >
          <div
            class="MuiBox-root MuiBox-root-61 makeStyles-headerAutoPay-52 makeStyles-row-12"
          >
            <div
              class="MuiBox-root MuiBox-root-62"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </div>
            <p
              class="MuiTypography-root makeStyles-headerTitle-55 MuiTypography-body1"
            >
              Enroll in AutoPay
            </p>
            <div
              class="MuiBox-root MuiBox-root-63"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-infoButton-54"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-64"
          >
            <div
              class="MuiDialogContent-root makeStyles-dialogContent-58"
            >
              <div
                class="MuiBox-root MuiBox-root-194 makeStyles-fauxInputField-69 makeStyles-fauxInputFieldDisabled-70"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
                >
                  Recurring Payment Amount
                </p>
                <p
                  class="MuiTypography-root makeStyles-fauxFloatRight-72 MuiTypography-body1"
                />
                <br />
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
                />
              </div>
              <div
                class="MuiBox-root MuiBox-root-76 makeStyles-fauxInputField-69 makeStyles-fauxInputFieldDisabled-70"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
                >
                  Payment Frequency
                </p>
                <p
                  class="MuiTypography-root makeStyles-fauxFloatRight-72 MuiTypography-body1"
                />
                <br />
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
                />
                <br />
                <p
                  class="MuiTypography-root makeStyles-quaternary-75 MuiTypography-body1 MuiTypography-colorTextSecondary"
                >
                  Next payment will be made on May 18, 2021
                </p>
              </div>
              <div
                class="MuiBox-root MuiBox-root-195 makeStyles-fauxInputField-69"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
                >
                  Pay From
                </p>
                <p
                  class="MuiTypography-root makeStyles-fauxFloatRight-72 MuiTypography-body1"
                />
                <br />
                <p
                  class="MuiTypography-root makeStyles-addAccount-41 MuiTypography-body1 MuiTypography-colorTextPrimary"
                >
                  Add Account
                </p>
                <button
                  class="MuiButtonBase-root MuiIconButton-root WithStyles(ForwardRef(IconButton))-root-196 makeStyles-iconButtonFilled-73 MuiIconButton-sizeSmall"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            <p
              class="MuiTypography-root makeStyles-helperText-39 MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              While enrolled in AutoPay, we'll automatically charge your payment amount on the due date. Your payment amount and payment date reflect your billing statements.
            </p>
            <p
              class="MuiTypography-root makeStyles-helperText-39 MuiTypography-body2 MuiTypography-colorTextSecondary"
            >
              There is a $15 fee for any ACH debit not honored by your bank
              .
            </p>
            <div
              class="MuiBox-root MuiBox-root-197 makeStyles-confirmingBox-38"
            />
          </div>
        </div>
        <div
          data-test="sentinelEnd"
          tabindex="0"
        />
      </div>
    </div>
    <div
      aria-hidden="true"
      class="MuiBackdrop-root makeStyles-confirmPanelBackdrop-93"
      style="opacity: 0; visibility: hidden;"
    />
    <div
      aria-hidden="true"
      class="MuiBackdrop-root makeStyles-confirmPanelBackdrop-104"
      style="opacity: 0; visibility: hidden;"
    />
    <div
      aria-hidden="true"
      class="MuiBackdrop-root makeStyles-confirmPanelBackdrop-116"
      style="opacity: 0; visibility: hidden;"
    />
    <div
      aria-hidden="true"
      class="MuiBackdrop-root makeStyles-confirmPanelBackdrop-120"
      style="opacity: 0; visibility: hidden;"
    />
    <div
      aria-hidden="true"
      class="MuiBackdrop-root makeStyles-confirmPanelBackdrop-132"
      style="opacity: 0; visibility: hidden;"
    />
    <div
      aria-hidden="true"
      class="MuiDrawer-root MuiDrawer-modal makeStyles-confirmPanelRoot-143"
      role="presentation"
      style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px; visibility: hidden;"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root"
        style="opacity: 0; visibility: hidden;"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorBottom makeStyles-modalPaper-33 MuiPaper-elevation16"
        style="visibility: hidden; webkit-transform: translateY(768px) translateY(0px); transform: translateY(768px) translateY(0px);"
        tabindex="-1"
      >
        <div
          class="MuiBox-root MuiBox-root-152 makeStyles-header-148 makeStyles-row-12"
        >
          <div
            class="MuiBox-root MuiBox-root-153 makeStyles-arrowBackBox-150"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
          </div>
          <p
            class="MuiTypography-root makeStyles-headerTitle-149 MuiTypography-body1"
          >
            New Payment Account
          </p>
          <div
            class="MuiBox-root MuiBox-root-154"
          />
        </div>
        <hr
          class="MuiDivider-root"
        />
        <div
          class="MuiDialogContent-root"
        >
          <div
            class="MuiFormControl-root makeStyles-inputField-139 MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputLabel-140 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
              data-shrink="true"
            >
              Routing Number
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root makeStyles-newPaymentMethodInput-141 MuiInputBase-fullWidth MuiInputBase-formControl"
              variant="outlined"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="routingNum"
                placeholder="Routing Number"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-155 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-157"
                >
                  <span>
                    Routing Number
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root makeStyles-inputField-139 MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputLabel-140 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
              data-shrink="true"
            >
              Account Number
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root makeStyles-newPaymentMethodInput-141 MuiInputBase-fullWidth MuiInputBase-formControl"
              variant="outlined"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="accountNum"
                placeholder="Account Number"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-155 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-157"
                >
                  <span>
                    Account Number
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root makeStyles-inputField-139 MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputLabel-140 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
              data-shrink="true"
            >
              Verify Account Number
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root makeStyles-newPaymentMethodInput-141 MuiInputBase-fullWidth MuiInputBase-formControl"
              variant="outlined"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                id="accountNumVerify"
                placeholder="Verify Account Number"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-155 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-157"
                >
                  <span>
                    Verify Account Number
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root makeStyles-inputFieldLast-144 MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputFieldLastLabel-145 makeStyles-inputLabel-140 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
              data-shrink="true"
            >
              Account Type
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <div
                aria-haspopup="listbox"
                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                role="button"
                tabindex="0"
              >
                <span>
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput"
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-155 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-156"
                  style="width: 0.01px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiPaper-root MuiCard-root makeStyles-saveAsDefaultCard-146 MuiPaper-elevation0 MuiPaper-rounded"
          >
            <div
              class="MuiBox-root MuiBox-root-159 makeStyles-saveAsDefaultBox-147"
            >
              <p
                class="MuiTypography-root makeStyles-inputLabel-140 MuiTypography-body1"
              >
                Save as default payment account
              </p>
              <span
                class="MuiSwitch-root"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-160 MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-checked-161 Mui-checked"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <input
                      checked=""
                      class="PrivateSwitchBase-input-163 MuiSwitch-input"
                      type="checkbox"
                      value=""
                    />
                    <span
                      class="MuiSwitch-thumb"
                    />
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiSwitch-track"
                />
              </span>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-164"
          >
            <button
              class="MuiButtonBase-root MuiButton-root makeStyles-root-35 makeStyles-root-165 MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation Mui-disabled MuiButton-fullWidth Mui-disabled"
              data-testid="PrimaryButton"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Add New Account
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
    <div
      aria-hidden="true"
      class="MuiBackdrop-root makeStyles-confirmPanelBackdrop-169"
      style="opacity: 0; visibility: hidden;"
    />
  </div>
</div>
`;
